kernelci/db/kernelci_api.py

```diff
@@ -88,10 +88,32 @@ def get_node(self, node_id):
         resp = self._get('/'.join(['node', node_id]))
         return json.loads(resp.text)
 
-    def get_nodes(self, attributes: dict = None):
+    def get_nodes(self, attributes: dict = None,
+                  offset: int = None, limit: int = None):
         """Get all nodes matching attributes"""
-        resp = self._get('nodes', params=attributes)
-        return resp.json()['items']
+        params = attributes.copy() if attributes else {}
+
+        if any((offset, limit)):
+            params.update({
+                'offset': offset or 0,
+                'limit': limit or 0,
```

Comment:
The limit must be greater than 0.

Reply:
Well I think the parameters should actually only be added to the data if they're not `None`.  I'll rework that.

Reply:
OK I've changed it to `or None` as that means the parameter won't be in the data if it's not provided, and the API will be using its own default value.